File: tools/followers.py
#!/usr/bin/env python3
"""List the accounts that follow a Bluesky user (walks the follow graph inward).

    python3 followers.py alice.bsky.social --limit 50
"""
import argparse
import sys

from actor_index import print_actors
from bsky import BskyError, get


def main() -> None:
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("actor", help="handle or DID whose followers to list")
    parser.add_argument("--limit", type=int, default=50)
    args = parser.parse_args()
    data = get("app.bsky.graph.getFollowers", {"actor": args.actor, "limit": args.limit})
    print_actors(data.get("followers", []))


if __name__ == "__main__":
    try:
        main()
    except BskyError as e:
        print(f"followers fetch failed: {e}", file=sys.stderr)
        sys.exit(1)

File: agent/.opencode/tools/followers.ts
import { tool } from "@opencode-ai/plugin"
import { $ } from "bun"

// Thin wrapper around tools/followers.py — located via the git worktree.
export default tool({
  description:
    "List the accounts that follow a Bluesky user (walks the follow graph " +
    "inward). Returns one line per account: `@handle (Name) — did · bio`. " +
    "Use it to see a user's audience or find who's connected to them.",
  args: {
    actor: tool.schema.string().describe("Handle or DID whose followers to list"),
    limit: tool.schema.number().describe("Max accounts to return").default(50),
  },
  async execute(args, context) {
    const script = `${context.worktree}/tools/followers.py`
    const limit = Number.isFinite(args.limit) ? Math.trunc(args.limit as number) : 50
    return await $`python3 ${script} ${args.actor} --limit ${String(limit)}`.env({ ...process.env, ACTING_SESSION: context.sessionID }).text()
  },
})
